test/realm_test.dart

```diff
@@ -552,5 +552,135 @@ Future<void> main([List<String>? args]) async {
+
+      var teams = realm.all<Team>();
+      expect(teams.length, 2);
+
+      //Convert RealmResults to List (asList should be implemented)
+      List<Team> list = [];
+      int itemsCount = teams.length;
+      for (var i = 0; i < itemsCount; i++) {
+        list.add(teams[i]);
+      }
+   
+      realm.write(() => realm.removeMany(list));
+      teams = realm.all<Team>();
+      expect(teams.length, 0);
+    });
+
+    test('Realm RemoveMany from RealmList', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create a Team
+      final team = Team()..name = "Ferrari";
+      realm.write(() => realm.add(team));
+
+      //Add players to the team
+      realm.write(() => team.players.addAll([
+            Person()..name = "Michael Schumacher",
+            Person()..name = "Sebastian Vettel",
+            Person()..name = "Kimi Räikkönen"
+          ]));
+
+      //Ensure the Tema exists in DB
+      var teams = realm.all<Team>();
+      expect(teams.length, 1);
+
+      //Remove team players
+      realm.write(() => realm.removeMany(teams[0].players));
+
+      //Ensure persons are deleted from DB
+      final personsFromDB = realm.all<Person>();
+      expect(personsFromDB.length, 0);
+    });
+
+    test('Realm RemoveMany from RealmList referenced by two objects', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create two Teams
+      final teamOne = Team()..name = "Ferrari";
+      final teamTwo = Team()..name = "Maserati";
+      realm.write(() => {realm.add(teamOne), realm.add(teamTwo)});
+
+      //Create common players list for both Teams
+      List<Person> players = [
+        Person()..name = "Michael Schumacher",
+        Person()..name = "Sebastian Vettel",
+        Person()..name = "Kimi Räikkönen"
+      ];
+      realm.write(() =>
+          {teamOne.players.addAll(players), teamTwo.players.addAll(players)});
+
+      //Ensule teams exist in DB
+      var teams = realm.all<Team>();
+      expect(teams.length, 2);
+
+      //Remove all players in a team
+      realm.write(() => realm.removeMany(teams[0].players));
+
+      //Ensure all persons are deleted from DB
+      final personsFromDB = realm.all<Person>();
+      expect(personsFromDB.length, 0);
+    });
+
+    test('Realm RemoveMany from RealmResults', () {
+      var config = Configuration([Team.schema, Person.schema]);
+      var realm = Realm(config);
+
+      //Create two Teams
+      final teamOne = Team()..name = "Ferrari";
+      final teamTwo = Team()..name = "Maserati";
+      realm.write(() => {realm.add(teamOne), realm.add(teamTwo)});
+
+      //Ensule teams exist in DB
+      var teams = realm.all<Team>();
+      expect(teams.length, 2);
+
+      //Remove all objects in RealmResults from DB
+      realm.write(() => realm.removeMany(teams));
+      teams = realm.all<Team>();
```

Comment:
Since it is a test I wanted to read them all again from DB. Just wanted to be sure that they are really deleted from DB, not only from the collection. Is there better way to do it? Do you think it is useless?

Reply:
I think there's value in both tests - i.e. ensuring that the collection has automatically updated as well as that the value in the database has been updated. Something like:

```dart
       //Remove all objects in RealmResults from DB
       realm.write(() => realm.removeMany(teams));
       
       // Ensure teams is now empty
       expect(teams.length, 0);

       // Just in case, try to re-read the table from the database and confirm its empty
       teams = realm.all<Team>();
       expect(teams.length, 0);
```

Reply:
Done.